Modified Wording for Key Attestations (#217)
openid/OpenID4VC-HAIP: openid4vc-high-assurance-interoperability-profile-1_0.md

### Key Attestation {#key-attestation}

> Wallets MUST support key attestations. If interoperability on the level of key attestations is required, such key attestations MUST conform to the definition in Annex D of [@!OIDF.OID4VCI], otherwise other key attestations formats can be used. 

Reviewer: The second MUST in this paragraph doesn't work. It would mean that compliance to this part of HAIP depends on whether "interoperability on the level of key attestations is required" which is an issue because:
- Conceptually, if we have a condition for a requirement that's external to the document, being compliant to HAIP would depend on the context where the wallet is used. This makes it vague what it makes to be compliant to HAIP, but also is pretty awkward for testing and certification.
- "interoperability on the level of key attestation is required" can be ambiguous. Which if it's connected to a MUST is a problem, much less so if it's a can, MAY etc.

Some options:

Wallets MUST support key attestations. It is not required for the wallet to support a specific format for key attestations. If interoperability on the level of key attestations is required, such key attestations SHOULD conform to the definition in Annex D of [@!OIDF.OID4VCI],  other key attestations formats MAY be used.

Wallets MUST support key attestations. Annex D of [@!OIDF.OID4VCI] specifies a standardized format for interoperability for key attestations which SHOULD be used. Other key attestation formats MAY be used.

Because of the SHOULD, maybe we should give examples of when to deviate, e.g. "Other key attestations formats MAY be used, for example because they provide different privacy or security guarantees in the context of the wallet implementation"

Reviewer 2: ```suggestion
It is RECOMMENDED that ecosystems that desire wallet-issuer interoperability on the level of key attestations require Wallets to support the format specified in Annex D of [@!OIDF.OID4VCI] instead. This may have an impact on security, privacy or availability due to the challenge presented in updating existing hardware key attestation formats.
```
applying gareth suggestion

`iat` and `exp` JWT claims express both the validity period of both the signature and the claims about the subject, unless there is a separate claim used to express the validity of the claims.

## Interoperable Key Attestations

Wallet implementations using the key attestation format specified in Annex D of [@!OIDF.OID4VCI] might need to utilize a transformation (backend) service to create such attestations based on data as provided in other formats by the respective platform or secure key management module. The dependency on such a service might impact the availability of the wallet app as well as the performance of the issuance process. This could be mitigated by creating keys and obtaining the respective key attestations in advance. 

# Security Considerations {#security_considerations}

The security considerations in [@!OIDF.OID4VCI] and [@!OIDF.OID4VP] apply.

# Privacy Considerations

## Interoperable Key Attestations

> Wallet implementations using the key attestation format specified in Annex D of [@!OIDF.OID4VCI] might need to utilize a transformation (backend) service to create such attestations based on data as provided in other formats by the respective platform or secure key management module. Such a backend service MUST be designed in a way that it does not use the attestation data to track users, e.g. the service should be stateless and just perform the transformation of the attestation data without binding the process in any way to a unqiue user identifier.  

Reviewer 3: Discussed on today's WG - consensus around changing text to this: "MUST be designed considering the privacy of the users"

> Wallet implementations using the key attestation format specified in Annex D of [@!OIDF.OID4VCI] might need to utilize a transformation (backend) service to create such attestations based on data as provided in other formats by the respective platform or secure key management module. Such a backend service MUST be designed in a way that it does not use the attestation data to track users, e.g. the service should be stateless and just perform the transformation of the attestation data without binding the process in any way to a unqiue user identifier.  

Reviewer 4: Would people be fine with this then?
```suggestion
Wallet implementations using the key attestation format specified in Annex D of [@!OIDF.OID4VCI] might need to utilize a transformation (backend) service to create such attestations based on data as provided in other formats by the respective platform or secure key management module. Such a backend service MUST be designed considering the privacy of its users. For example, the service could be stateless and just perform the transformation of the attestation data without binding the process in any way to a unique user identifier.
```

> Wallet implementations using the key attestation format specified in Annex D of [@!OIDF.OID4VCI] might need to utilize a transformation (backend) service to create such attestations based on data as provided in other formats by the respective platform or secure key management module. Such a backend service MUST be designed in a way that it does not use the attestation data to track users, e.g. the service should be stateless and just perform the transformation of the attestation data without binding the process in any way to a unqiue user identifier.  

Author: I like your proposal @c2bo .
